src/neo4j_graphrag/utils.py

```diff
@@ -14,11 +14,26 @@
 #  limitations under the License.
 from __future__ import annotations
 
-from typing import Optional
+import inspect
+from typing import Any, Optional, Union
+
+import neo4j
 
 
 def validate_search_query_input(
     query_text: Optional[str] = None, query_vector: Optional[list[float]] = None
 ) -> None:
     if not (bool(query_vector) ^ bool(query_text)):
         raise ValueError("You must provide exactly one of query_vector or query_text.")
+
+
+async def execute_query(
+    driver: Union[neo4j.Driver, neo4j.AsyncDriver], query: str, **kwargs: Any
+) -> list[neo4j.Record]:
+    if inspect.iscoroutinefunction(driver.execute_query):
+        records, _, _ = await driver.execute_query(query, **kwargs)
+        return records  # type: ignore[no-any-return]
+    # ignoring type because mypy complains about coroutine
+    # but we're sure at this stage we do not have a coroutine anymore
+    records, _, _ = driver.execute_query(query, **kwargs)  # type: ignore[misc]
+    return records  # type: ignore[no-any-return]
```

Comment:
Why do we ignore types here?

Reply:
I see, happy to return to this in future

Reply:
Card created for better investigation